Update CMake to be more lenient with package requirements (#1436)
aws/aws-iot-device-sdk-embedded-C: demos/jobs/jobs_demo_mosquitto/README.md

    apt install curl libmosquitto-dev

> Version 1.4.10 of libmosquitto is required to run this demo.

Author: ```suggestion
libmosquitto 1.4.10 or any later version of the first major release is required to run this demo.
```

> To build the latest version of libmosquitto (1.6.12) for ALPN support,

Reviewer: Version 2.0.0. came out last week, so this is no longer accurate

> To build the latest version of libmosquitto (1.6.12) for ALPN support,

Author: ```suggestion
For ALPN support, build the latest version of the first major release of libmosquitto (1.6.12).
```

> go to the libmosquitto directory,

Author: ```suggestion
To do this, go to the libmosquitto directory,
```
